### PR TITLE
feat: basic meltano superset utility extension

### PR DESCRIPTION
**Pull request by pandemicsyn**
## Example meltano yaml

```yaml
plugins:
  utilities:
  - name: superset_ext
    namespace: superset
    pip_url: apache-superset==2.0.0 flask==2.0.3 werkzeug==2.0.3 jinja2==3.0.1 wtforms==2.3.3
      -e /Users/syn/projects/ext-superset
    executable: superset_invoker
    commands:
      describe:
        args: describe
        executable: superset_extension
      initialize:
        args: initialize
        executable: superset_extension
      create_admin:
        args: create_admin
        executable: superset_extension
      ui: run --port 8088 --host 127.0.0.1
    config:
      ROW_LIMIT: 42
      SQLALCHEMY_DATABASE_URI: sqlite:///$MELTANO_PROJECT_ROOT/superset/superset.db
      SUPERSET_WEBSERVER_PORT: 8088
      SUPERSET_CONFIG_PATH: $MELTANO_PROJECT_ROOT/superset/superset_config.py
      SECRET_KEY: OaN2xDksQXXygKmlLINUNxIKMWE3z2WK2gtJEdRaFYkREEiialUo/+Um
      FLASK_APP: superset
```

## Sample setup

Starting with just a meltano.yml:

```
(ext-dev) ➜  extdev tree
.
└── meltano.yml

0 directories, 1 file
```

Installing the extension and performing the initialization steps:

```
(ext-dev) ➜  extdev meltano install utility superset_ext
Installing 1 plugins...
Installing utility 'superset_ext'...
Installed utility 'superset_ext'
(ext-dev) ➜  extdev rm -rf .meltano/utilities/superset_ext
(ext-dev) ➜  extdev clear
(ext-dev) ➜  extdev export FLASK_APP=superset
(ext-dev) ➜  extdev clear
(ext-dev) ➜  extdev tree
.
└── meltano.yml

0 directories, 1 file
(ext-dev) ➜  extdev meltano install utility superset_ext
Installing 1 plugins...
Installing utility 'superset_ext'...
Installed utility 'superset_ext'
(ext-dev) ➜  extdev meltano invoke superset_ext:initialize
Superset initialized, don't forget to configure an admin user!
(ext-dev) ➜  extdev meltano config superset_ext set SECRET_KEY $(openssl rand -base64 42)
Utility 'superset_ext' setting 'SECRET_KEY' was set in `meltano.yml`: 'OaN2xDksQXXygKmlLINUNxIKMWE3z2WK2gtJEdRaFYkREEiialUo/+Um'
(ext-dev) ➜  extdev meltano invoke superset_ext:create_admin --username=syn
Firstname [admin]: me
Lastname [user]: sure
Email [admin@superset]: me@human.com
Password:
Repeat for confirmation:
## output from here on is from Superset itself ##
logging was configured successfully
2022-09-08 17:27:49,665:INFO:superset.utils.logging_configurator:logging was configured successfully
2022-09-08 17:27:49,668:INFO:root:Configured event logger of type <class 'superset.utils.log.DBEventLogger'>
Falling back to the built-in cache, that stores data in the metadata database, for the following cache: `FILTER_STATE_CACHE_CONFIG`. It is recommended to use `RedisCache`, `MemcachedCache` or another dedicated caching backend for production deployments
2022-09-08 17:27:49,669:WARNING:superset.utils.cache_manager:Falling back to the built-in cache, that stores data in the metadata database, for the following cache: `FILTER_STATE_CACHE_CONFIG`. It is recommended to use `RedisCache`, `MemcachedCache` or another dedicated caching backend for production deployments
Falling back to the built-in cache, that stores data in the metadata database, for the following cache: `EXPLORE_FORM_DATA_CACHE_CONFIG`. It is recommended to use `RedisCache`, `MemcachedCache` or another dedicated caching backend for production deployments
2022-09-08 17:27:49,670:WARNING:superset.utils.cache_manager:Falling back to the built-in cache, that stores data in the metadata database, for the following cache: `EXPLORE_FORM_DATA_CACHE_CONFIG`. It is recommended to use `RedisCache`, `MemcachedCache` or another dedicated caching backend for production deployments
No PIL installation found
2022-09-08 17:27:49,726:INFO:superset.utils.screenshots:No PIL installation found
Loaded your LOCAL configuration at [/Users/syn/projects/meltano-projects/extdev/superset/superset_config.py]
Recognized Database Authentications.
Admin User syn created.
```

Resulting fs layout:

```
(ext-dev) ➜  extdev tree
.
├── meltano.yml
└── superset
    ├── __pycache__
    │   └── superset_config.cpython-38.pyc
    ├── superset.db
    └── superset_config.py

2 directories, 4 files
```

Resulting config:

```python
(ext-dev) ➜  extdev cat superset/superset_config.py
import sys
module = sys.modules[__name__]
config = {'ROW_LIMIT': '42', 'SQLALCHEMY_DATABASE_URI': 'sqlite:////Users/syn/projects/meltano-projects/extdev/superset/superset.db', 'SUPERSET_WEBSERVER_PORT': '8088', 'SUPERSET_CONFIG_PATH': '/Users/syn/projects/meltano-projects/extdev/superset/superset_config.py', 'SECRET_KEY': 'OaN2xDksQXXygKmlLINUNxIKMWE3z2WK2gtJEdRaFYkREEiialUo/+Um', 'FLASK_APP': 'superset', 'PATH': '/Users/syn/projects/meltano-projects/extdev/.meltano/utilities/superset_ext/venv/bin:/Users/syn/.pyenv/versions/ext-dev/bin:/opt/homebrew/Cellar/pyenv/2.3.2/libexec:/opt/homebrew/Cellar/pyenv/2.3.2/plugins/python-build/bin:/Users/syn/google-cloud-sdk/bin:/opt/homebrew/opt/openjdk/bin:/opt/homebrew/opt/mysql-client/bin:/Users/syn/perl5/bin:/Users/syn/.nvm/versions/node/v16.13.2/bin:/Users/syn/.pyenv/shims:/Users/syn/.pyenv/bin:/opt/homebrew/Cellar/pyenv-virtualenv/1.1.5/shims:/Users/syn/bin:/usr/local/bin:/Users/syn/.pyenv/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin:/usr/local/MacGPG2/bin:/Library/Apple/usr/bin:/Users/syn/.cargo/bin'}
for key, value in config.items():
    if key.isupper():
        setattr(module, key, value)

# add any additional config here # 

```

closes #1

### CODE REVIEW

**Comment by pandemicsyn** [comment]
@aaronsteers @pnadolny13 is there anything we wanna do to make setup a little simpler for users in this new version? i.e. automatically generate a secret key via `meltano config superset_ext set SECRET_KEY $(openssl rand -base64 42)` for the user if we see one isn't set?

File: pyproject.toml
Comment (pandemicsyn) [inline pyproject.toml:60]: no static assets for superset required (yet) 🥳

Comment (pnadolny13) [reply pyproject.toml:60]: @pandemicsyn we debated about something like secret key generation during the original implementation but I think the sentiment was that we'd prefer to tell users to run a few explicit commands vs being overly smart under the hood.

My personal view is that in some cases we should otherwise were simply wrapping the existing commands which doesnt end up adding a lot of value. But it seems like thats a debate for a future iteration. I think sticking with the status quo for this iteration is fine.

File: README.md
Comment (pandemicsyn) [inline README.md]: will need to bump this before merge to point to main.

**Comment by pandemicsyn** [comment]
ok , think this is ready for review

**Comment by aaronsteers** [comment]
> @aaronsteers @pnadolny13 is there anything we wanna do to make setup a little simpler for users in this new version? i.e. automatically generate a secret key via `meltano config superset_ext set SECRET_KEY $(openssl rand -base64 42)` for the user if we see one isn't set?

@pandemicsyn - Sorry for the delay in my reply. I may need additional context to give a helpful reply here. If we need to add this in order to get "parity", then I'd say that is certainly let's see what we can do to keep parity - or else add onboarding docs to fill the gap.

If we don't need this for parity, I think it is probably best to move into a follow-on action. My concern is _where_ and _how_ the key should be stored. We'd probably need to ensure the default location is protected from being accidentally commited via `.gitignore`, and with that comes the problem of the key not being portable in deployment scenarios.

**Comment by pandemicsyn** [comment]
> If we need to add this in order to get "parity"

We don't need this for parity, it was just an example of something we might do. The core of my question was basically "do we just want parity or is there anything we want to improve upon". This question pre-dates the dbt convo we had, but based on the dbt convo I'd already assumed status quo for Superset was fine for this first version 😄  So It's safe to ignore that question.

File: README.md
Comment (pnadolny13) [inline README.md]: @pandemicsyn I'm getting `FileNotFoundError: [Errno 2] No such file or directory: '/Users/pnadolny/Documents/Git/meltano_project/sdk_cookie_test/test_123/superset/superset_config.py'`.

I have this meltano.yml:

```yaml
plugins:
  utilities:
  - name: superset
    namespace: superset
    pip_url: apache-superset==2.0.0 flask==2.0.3 werkzeug==2.0.3 jinja2==3.0.1 wtforms==2.3.3 
        git+https://github.com/meltano/superset-ext.git@feat/init-extension-take2
    executable: superset_invoker
    commands:
      describe:
        args: describe
        executable: superset_extension
      initialize:
        args: initialize
        executable: superset_extension
      create_admin:
        args: create_admin
        executable: superset_extension
      ui: run --port 8088 --host 127.0.0.1
    config:
      ROW_LIMIT: 42
      SQLALCHEMY_DATABASE_URI: sqlite:///$MELTANO_PROJECT_ROOT/superset/superset.db
      SUPERSET_WEBSERVER_PORT: 8088
      SUPERSET_CONFIG_PATH: $MELTANO_PROJECT_ROOT/superset/superset_config.py
      SECRET_KEY: OaN2xDksQXXygKmlLINUNxIKMWE3z2WK2gtJEdRaFYkREEiialUo/+Um
      FLASK_APP: superset
```

And I ran:

```bash
meltano install utility superset
meltano invoke superset:initialize
```

Was it supposed to create that file for me or did I miss a step?

**Review by pnadolny13 — CHANGES_REQUESTED** [review]

File: README.md
Comment (pnadolny13) [reply README.md]: @pandemicsyn after our zoom chat these issues are no longer relevant. Everything else looked good to me! I got the samples loaded and everything worked, no permissions issues.

I would make one note that the current superset hub instructions have a command `create-admin` with a dash and this one is and underscore.

Comment (pandemicsyn) [reply README.md]: easy-peasy , updated to use `create-admin`

**Review by tayloramurphy — APPROVED** [review]

**Review by pnadolny13 — APPROVED** [review]
@pandemicsyn @tayloramurphy LGTM!

**Comment by tayloramurphy** [comment]
@aaronsteers I'm going to go ahead and merge this. Functionally it works and has approvals from myself and Pat. It's also a blocker for the Hub PR which I want us to get merged too.

**Pull request merged**
